src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -148,7 +148,10 @@ public function checkRequestPath(Request $request, string $path)
      */
     public function generateUri(Request $request, string $path)
     {
-        if (str_starts_with($path, 'http') || !$path) {
+        $url = parse_url($path);
+
+        // absolute URL
+        if ('' === $path || isset($url['scheme'], $url['host'])) {
```

Comment:
`'' === $path` is not really an absolute URL. The comment is misleading.

Reply:
comment removed